### PR TITLE
RSDK-1229 RSDK-2693 Remove robot from motion service constructor, and prepare for removal from framesystem service constructor

### PR DESCRIPTION
**Pull request by biotinker**
Note that this also alters pointcloud as that uses the FS service and was affected by these changes; added @bhaney to look at that, feel free to swap to someone else if needed.

The design of this update removes any need for Motion to talk to the robot or individual resources in real-time, and moves all responsibility of keeping track of discrete resources to the frame system service.

### CODE REVIEW

**Review by bhaney — APPROVED** [review]
Cool! Thanks for removing the robot dependency

**Comment by edaniels** [comment]
woo!

File: robot/framesystem/framesystem.go
Comment (raybjork) [inline robot/framesystem/framesystem.go:235]: [nit] This error shows up in a couple places, can we make it a named function in this package?

File: services/motion/builtin/builtin.go
Comment (raybjork) [inline services/motion/builtin/builtin.go:45]: [nit] Can this comment on this function indicate that it returning the dependencies?

File: services/motion/builtin/builtin.go
Comment (raybjork) [inline services/motion/builtin/builtin.go:80]: [nit] redundant line

File: services/motion/builtin/builtin.go
Comment (raybjork) [inline services/motion/builtin/builtin.go:190]: [minor] Can't this all just reduce down to `arm.FromRobot(componentName.ShortName()]`?

File: services/motion/builtin/builtin.go
Comment (raybjork) [inline services/motion/builtin/builtin.go:234]: [opt] can merge with if statement below

**Review by raybjork — CHANGES_REQUESTED** [review]

File: motionplan/motionPlanner.go
Comment (biotinker) [reply motionplan/motionPlanner.go:75]: We can, but that turns a no-output change into a breaking change

File: robot/framesystem/framesystem.go
Comment (biotinker) [reply robot/framesystem/framesystem.go:216]: All specified transforms are static, so I'm not sure how that would affect this. You can't ephemerally specify something that is InputEnabled.

File: motionplan/motionPlanner.go
Comment (biotinker) [reply motionplan/motionPlanner.go:75]: Not even, just changes the signature for a public function. I think you're right, it's likely worthwhile.

File: robot/framesystem/framesystem.go
Comment (raybjork) [reply robot/framesystem/framesystem.go:216]: Right, we're just returning the inputs and not the poses.  This makes sense thanks

File: robot/framesystem/framesystem.go
Comment (spieswl) [inline robot/framesystem/framesystem.go:47]: Fantastic

File: motionplan/motionPlanner.go
Comment (biotinker) [reply motionplan/motionPlanner.go:75]: Actually no, since other methods have their seed map passed in. This is the only one using `robot`, and it's just a convenience function that isn't used.

File: robot/framesystem/framesystem.go
Comment (tessavitabile) [reply robot/framesystem/framesystem.go:223]: Is allOriginals used for anything?

Comment (biotinker) [reply robot/framesystem/framesystem.go:223]: Looking at this more closely; no, not anymore. Will remove.

File: services/motion/builtin/builtin.go
Comment (spieswl) [inline services/motion/builtin/builtin.go:63]: [nit] Comment description

File: services/motion/builtin/builtin.go
Comment (spieswl) [inline services/motion/builtin/builtin.go:219]: [minor] Could `fsInputs` be assigned on L189 and used here to cut down on a call to `AllCurrentInputs`?

**Review by spieswl — CHANGES_REQUESTED** [review]
Modulo function signature changes and some docs cleanup, not much else to be fixed up here.

File: services/motion/builtin/builtin.go
Comment (biotinker) [reply services/motion/builtin/builtin.go:190]: No, because there is no `Robot` to

Comment (raybjork) [reply services/motion/builtin/builtin.go:188]: Ah yes I see now.  Still can be a little less verbose if you directly index the resources map with the `componentName.ShortName()` and cast it to an arm.

File: components/arm/xarm/xarm.go
Comment (raybjork) [inline components/arm/xarm/xarm.go:163]: [q] why is this being added? When will this be true?

File: components/arm/xarm/xarm.go
Comment (raybjork) [inline components/arm/xarm/xarm.go:175]: I think if you're going to do this you should add a `<=` check on L148

File: components/arm/xarm/xarm.go
Comment (biotinker) [reply components/arm/xarm/xarm.go:163]: conn is nil on the initial reconfigure that initializes the arm at robot startup.

conn is not nil when the arm config gets changed.

File: robot/framesystem/framesystem.go
Comment (raybjork) [inline robot/framesystem/framesystem.go:334]: This feels like an improvement but would be a pretty big change from current behavior, right?

File: components/arm/xarm/xarm.go
Comment (biotinker) [reply components/arm/xarm/xarm.go:175]: I think the better thing to do there is error on negative speed, not overwrite it.

**Review by raybjork — CHANGES_REQUESTED** [review]

File: components/arm/xarm/xarm.go
Comment (raybjork) [reply components/arm/xarm/xarm.go:163]: Ok I see, the if statement at L156 got me confused, but yes this makes sense

File: components/arm/xarm/xarm.go
Comment (raybjork) [reply components/arm/xarm/xarm.go:175]: Yeah this is better I think

File: robot/framesystem/framesystem.go
Comment (biotinker) [reply robot/framesystem/framesystem.go:334]: It's not actually a behavior change in terms of meaningful results. Without this change, the FS service spouts errors until the resource is available, at which point everything is fine. This way, those errors are silenced, but because the only way for this to happen is for a resource to be in the resource graph but error on actual retrieval, conditions for errors here only occur briefly during startup.

Making the FS service tolerant of not-yet-initialized resources was Eric's suggested solution here.

The only behavioral difference this causes is the suppression of these spurious error messages. Both before and now, an attempt to use an uninitialized resource will fail. Manual testing with remotes and such yielded no issues.

Comment (biotinker) [reply robot/framesystem/framesystem.go:334]: Shorter answer, this changes no actual *behavior*, will just mildly change which error message a user will get in certain scenarios.

Comment (raybjork) [reply robot/framesystem/framesystem.go:334]: Sounds great, I'm all for this

Comment (spieswl) [reply robot/framesystem/framesystem.go:334]: I think the worst case scenario is that a component actually doesn't have ModelFrame information but is skipped and the user isn't aware until they try to move the component. I feel like we should log a warning for this but continue with `continue`

File: robot/framesystem/framesystem.go
Comment (spieswl) [inline robot/framesystem/framesystem.go:233]: [q] Was this specified somewhere and I missed it?

Comment (biotinker) [reply robot/framesystem/framesystem.go:233]: ...I forgot to add it to git :P

File: robot/framesystem/framesystem.go
Comment (biotinker) [reply robot/framesystem/framesystem.go:334]: Anything that would normally have model frame information but does not, should error all on its own during its own initialization.

File: robot/framesystem/framesystem.go
Comment (spieswl) [reply robot/framesystem/framesystem.go:233]: :P

**Review by spieswl — APPROVED** [review]
LGTM, nice work!

**Comment by tessavitabile** [comment]
No need to wait for my sign-off on this

File: services/motion/builtin/builtin.go
Comment (biotinker) [reply services/motion/builtin/builtin.go:188]: Done

**Review by raybjork — APPROVED** [review]

**Pull request merged**
